evcouplings/compare/hmmer_wrappers.py

```diff
@@ -0,0 +1,309 @@
+        Inclusion threshold applied on the domain level
+        (e.g. "1E-03" or 0.001 or 50)
+    seq_threshold : int or float or str
+        Inclusion threshold applied on the sequence level
+        (e.g. "1E-03" or 0.001 or 50)
+    nobias : bool, optional (default: False)
+        Turn of bias correction
+    cpu : int, optional (default: None)
+        Number of CPUs to use for search. Uses all if None.
+    stdout_redirect : str, optional (default: None)
+        Redirect bulky stdout instead of storing
+        with rest of results (use "/dev/null" to dispose)
+    binary : str (default: "hmmsearch")
+        Path to jackhmmer binary (put in PATH for
+        default to work)
+
+    Returns
+    -------
+    HmmsearchResult
+        namedtuple with fields corresponding to the different
+        output files (prefix, alignment, output, tblout, domtblout)
+
+    Raises
+    ------
+    ExternalToolError, ResourceError
+    """
+    verify_resources(
+        "Input file does not exist or is empty",
+        hmmfile, database
+    )
+
+    create_prefix_folders(prefix)
+
+    # store filenames of all individual results;
+    # these will be returned as result of the
+    # function.
+    result = HmmsearchResult(
+        prefix,
+        prefix + ".sto",
+        prefix + ".output" if stdout_redirect is None else stdout_redirect,
+        prefix + ".tblout",
+        prefix + ".domtblout"
+    )
+
+    cmd = [
+        binary,
+        "-o", result.output,
+        "-A", result.alignment,
+        "--tblout", result.tblout,
+        "--domtblout", result.domtblout,
+        "--noali",
+        "--notextw"
+    ]
+
+    # reporting thresholds are set accordingly to
+    # inclusion threshold to reduce memory footprit
+    if use_bitscores:
+        cmd += [
+            "-T", str(seq_threshold),
+            "--domT", str(domain_threshold),
+            "--incT", str(seq_threshold),
+            "--incdomT", str(domain_threshold)
+        ]
+    else:
+        cmd += [
+            "-E", str(seq_threshold),
+            "--domE", str(domain_threshold),
+            "--incE", str(seq_threshold),
+            "--incdomE", str(domain_threshold)
+        ]
+
+    # number of CPUs
+    if cpu is not None:
+        cmd += ["--cpu", str(cpu)]
+
+    # bias correction filter
+    if nobias:
+        cmd += ["--nobias"]
+
+    cmd += [hmmfile, database]
+
+    return_code, stdout, stderr = run(cmd)
+
+    # also check we actually created some sort of alignment
+    verify_resources(
+        "hmmsearch returned empty alignment: "
+        "stdout={} stderr={} file={}".format(
+            stdout, stderr, result.alignment
+        ),
+        result.alignment
+    )
+
+    return result
+
+
+
+def hmmbuild_and_search(database, prefix,
+                  use_bitscores, domain_threshold, seq_threshold,
+                  nobias=False, cpu=None,
+                  stdout_redirect=None,
+                  binary=["hmmbuild", "hmmsearch"]):
```

Comment:
Have two separate parameters here (binary_hmmbuild and binary_hmmsearch).

General comment: never use lists as default arguments, this can cause nasty effects